Server/Controllers/MatchController.cs

```diff
@@ -275,6 +293,113 @@ public async Task<List<object>> Events([FromBody] MatchEventsRequest request)
+            var scoreDelta = CalculateScoreDelta(averageWinningRating, averageLosingRating);
+            await UpdateRating(winningTeam, losingTeam, scoreDelta);
+            return allPlayers;
+        }
+
+
+        /// <summary>
+        /// Formula is the difference between loosing and winning team of avg rating divided by 40
+        /// </summary>
+        /// <param name="averageWinningRating"></param>
+        /// <param name="averageLosingRating"></param>
+        /// <returns></returns>
+        private static int CalculateScoreDelta(double averageWinningRating, double averageLosingRating)
+        {
+            var scoreDeltaDouble =
+                -(averageWinningRating - averageLosingRating) /
+                DivisionPoints; // e.g. - (1900 - 2100) / 40 = 5. Meaning winning team will get 5 more points 
+                                // than the base as their average was weaker
+            int scoreDelta =
+                Convert.ToInt32(Math.Round(scoreDeltaDouble, 0,
+                    MidpointRounding.AwayFromZero)); // we don't do a conversion straight to the double
+                                                     // as it will do a MidpointRounding.ToEven by default
+            scoreDelta = scoreDelta > MaximumDelta ? MaximumDelta : scoreDelta;
+            return scoreDelta;
+        }
+
+        private static IEnumerable<Player> GetLosingTeam(AfterMatchRequest request, List<Player> allPlayers, out double averageLosingRating)
+        {
+            var losingTeamIds = request
+                .Players
+                .Where(p => p.Team != request.Winner)
+                .Select(p => ulong.Parse(p.SteamId))
+                .ToList();
+            var losingTeam = allPlayers.Where(p => losingTeamIds.Contains(p.SteamId));
+            averageLosingRating = losingTeam.Average(p => p.Rating);
+            return losingTeam;
+        }
+
+        private static IEnumerable<Player> GetWinningTeam(AfterMatchRequest request, List<Player> allPlayers, out double averageWinningRating)
+        {
+            var winningTeamIds = request
+                .Players
+                .Where(p => p.Team == request.Winner)
+                .Select(p => ulong.Parse(p.SteamId))
+                .ToList();
+            var winningTeam = allPlayers.Where(p => winningTeamIds.Contains(p.SteamId));
+            averageWinningRating = winningTeam.Average(p => p.Rating);
+            return winningTeam;
+        }
+
+        private async Task UpdateRating(IEnumerable<Player> winningTeam, IEnumerable<Player> losingTeam, int scoreDelta)
+        {
+            foreach (var player in winningTeam)
+                player.Rating = player.Rating + (BaseRating + scoreDelta);
+            foreach (var player in losingTeam)
+                player.Rating = player.Rating - (BaseRating + scoreDelta);
+            _context.UpdateRange(winningTeam);
+            _context.UpdateRange(losingTeam);
+            await _context.SaveChangesAsync();
```

Comment:
All changes in that action should be saved in a single transaction. Just move a call to `UpdateNewRating` in the action before it's SaveChanges call, and remove that one